echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/TriggerRepository.groovy

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.echo.scheduler.actions.pipeline.impl
+
+import com.netflix.spinnaker.echo.model.Pipeline
+import com.netflix.spinnaker.echo.model.Trigger
+import groovy.util.logging.Slf4j
+
+import javax.annotation.Nullable
+
+@Slf4j
+class TriggerRepository {
+  private final Map<String, Trigger> triggersById
+
+  TriggerRepository(List<Pipeline> pipelines) {
+    triggersById = new HashMap<>()
+    for (Pipeline pipeline : pipelines) {
+      for (Trigger trigger : pipeline.triggers) {
+        if (trigger.id == null) {
+          // see PipelineCache::decorateTriggers
+          throw new IllegalStateException("Trigger with null id ${trigger}")
```

Comment:
this may be too broad of an exception ... could you just throw a `NullPointerException` with the same message

Reply:
you've asserted elsewhere that triggers aren't going to have a null id so is this subsequent check really doing much?

Reply:
They shouldn't be null anymore coming out of `PipelineCache`, but if we there is ever a change that circumvents `decorateTriggers` and we are back to having null ids, it will break assumptions here very hard